### PR TITLE
Add factory reset after 5 reboots example to cookbook

### PR DESCRIPTION
**Pull request by nagyrobi**
taken from https://github.com/esphome/esphome/pull/5887#issue-2022819069 thanks @fornellas

## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: cookbook/lambda_magic.rst [hunk 1]
```diff
@@ -400,6 +400,77 @@ will return ``NaN``, which corresponds to ``unknown`` sensor state.
...
+    
+    globals:
+      - id: fast_boot
+        type: int
+        restore_value: yes
+        initial_value: '0'
+    
+      - id: factory_reset_boot_count_trigger
+        type: int
+        initial_value: '5'
```
Comment (jesserockz) [inline cookbook/lambda_magic.rst]: This doesn't really make sense as a global, it could just be a substitution

Comment (nagyrobi) [reply cookbook/lambda_magic.rst]: Done

**Comment by fornellas** [comment]
Ping @jesserockz for review.

File: cookbook/lambda_magic.rst [hunk 2]
```diff
@@ -400,6 +400,74 @@ will return ``NaN``, which corresponds to ``unknown`` sensor state.
...
+    
+    wifi:
+      id: wifi_component
+      ap:
+        ap_timeout: 0s
+      reboot_timeout: 0s
+    
+    captive_portal:
+    
+    switch:
```
Comment (jesserockz) [inline cookbook/lambda_magic.rst]:
```suggestion
    button:
```

File: cookbook/lambda_magic.rst [hunk 3]
```diff
@@ -400,6 +400,74 @@ will return ``NaN``, which corresponds to ``unknown`` sensor state.
...
+          - if:
+              condition:
+                lambda: return ( id(fast_boot) >= ${factory_reset_boot_count_trigger});
+              then:
+                - lambda: |-
+                    ESP_LOGD("Fast Boot Factory Reset", "Performing factotry reset");
+                    id(fast_boot) = 0;
+                    fast_boot->loop();
+                    global_preferences->sync();
+                - switch.turn_on: factory_reset_switch
```
Comment (jesserockz) [inline cookbook/lambda_magic.rst]:
```suggestion
                - button.press: factory_reset_button
```

File: cookbook/lambda_magic.rst [hunk 4]
```diff
@@ -400,6 +400,74 @@ will return ``NaN``, which corresponds to ``unknown`` sensor state.
...
+      id: wifi_component
+      ap:
+        ap_timeout: 0s
+      reboot_timeout: 0s
+    
+    captive_portal:
+    
+    switch:
+      - platform: factory_reset
+        id: factory_reset_switch
```
Comment (jesserockz) [inline cookbook/lambda_magic.rst]:
```suggestion
        id: factory_reset_button
```

**Review by jesserockz — DISMISSED** [review]

File: cookbook/lambda_magic.rst [hunk 5]
```diff
@@ -400,6 +400,74 @@ will return ``NaN``, which corresponds to ``unknown`` sensor state.
         name: "Number from text"
 
 
+Factory reset after 5 quick reboots
+-----------------------------------
+
+One may want to restore factory settings (eg. Wi-Fi credentials set at runtime, or clear restore states) without having to
+disassemble or dismount the devices from their deployed location, whilst there's no network access either. The example below
+shows how to achieve that using lambdas in a script by triggerig the factory reset switch after the system rebooted 5 times
+with 10 seconds timeframes.
```
Comment (frenck) [inline cookbook/lambda_magic.rst]:
```suggestion
One may want to restore factory settings (like Wi-Fi credentials set at runtime, or clear restore states) without having to
disassemble or dismount the devices from their deployed location, whilst there's no network access either. The example below
shows how to achieve that using lambdas in a script by triggering the factory reset switch after the system rebooted 5 times
with 10-second timeframes.
```

**Review by frenck — APPROVED** [review]
Thanks, @nagyrobi 👍 

../Frenck

**Pull request merged**